website/content/en/docs/best-practices/common-recommendation.md

```diff
@@ -0,0 +1,73 @@
+
+## Common Recommendations
+
+### Develop idempotent reconciliation solutions
+
+See that following the operator's pattern; you will create controllers, from [controller-runtime][controller-runtime], with a reconcile function responsible for synchronizing the resources until a desired state on the cluster. So, it is recommended to develop idempotent solutions for your reconciliations.
+
+### Understand the API's concept
+
+Not necessary but usually, the primary resource managed by these controllers which are reconciled are from your own APIs which [Extend the Kubernetes API with CustomResourceDefinitions][k8s-crd-doc]. So, check  documentation [Groups and Versions and Kinds, oh my!][kb-gkv] to have a better idea and be able to design better solutions.
+
+### Avoid a design solution where more than one Kind is reconciled by the same controller
+
+Have many Kinds, for example many CRDs, which are all managed by the same controller such as `install.go` usually goes against the design proposed by [controller-runtime][controller-runtime] beyond might hurt concepts such as; encapsulation, the Single Responsibility principle, and Cohesion. Damaging these concepts could cause unexpected side effects, such as; difficulty extending, to reuse, and low maintainability, only to mention a few.
+
+### Others common suggestions
+
+Additionally, we can share:
+
+- Provide the images and tags used by the operator solution via environment variables in the `config/manager/manager.yaml`: 
+
+```yaml
+...
+spec:
+  ...
+    spec:
+      ...
+      containers:
+      - command:
+        - /manager
+        ...
+        env:
+        - name: MY_IMAGE
+          value: "quay.io/example.com/image:0.0.1"
+```
+
+- Manage your solutions using [Status Conditionals][status-conditionals] 
+- Use [finalizers][finalizers] when/if required 
+- Cover the project with tests/CI to ensure its quality:
+    - For any language-based operator, you can use [Scorecard][scorecard] to implement functional tests
+    - For Go-based operators, you can also use [envtest][envtest] to cover the controllers. For further information see [Testing with EnvTest][testing-with-envtest]
+    - For Ansible-based operators, you can also use [Molecule][molecule]. For further information see [Testing with Molecule][molecule-tests]
+    - For Helm-based operators, you can also use [Chart tests][helm-chart-tests]
+- Ensure that you checked the [FAQ question Can I customize the projects generated with SDK tool?][faq] and understand the [Project Layout][project-layout] before start to do your customizations as please you on top.
+- If you will integrate your project with OLM then, also check its [Best Practices][olm-best-practices] section.
+ 
```

Comment:
A teammate has [written an article](https://www.openshift.com/blog/7-best-practices-for-writing-kubernetes-operators-an-sre-perspective) about some best practices for writing an operator (from an SRE perspective). Could be added as a reference in one of the bullet points, or in a separate section for further reading. WDYT?

Reply:
I think we need to stick with more official links. WDYT about after we get this merged he push a PR with suggested items that would fit here? and/or collab with https://operator-framework.github.io/community-operators/best-practices/ to make that better instead of creating another doc?

See that the idea here is to have what is more specific to sdk and then, we can link that one which is very generic.

Reply:
That sounds like a plan @camilamacedo86. SGTM.